src/supervisor.tex

```diff
@@ -2258,3 +2273,103 @@ \chapter{``Svnapot'' Standard Extension for NAPOT Translation Contiguity, Versio
   configurations result in page-faults exceptions, while invalid access
   types or accesses to invalid physical memory regions trigger page faults.
 \end{commentary}
+
+\chapter{``Svpbmt'' Standard Extension for Page-Based Memory Attributes, Version 0.1}
+\label{svpbmt}
+
+{\bf Warning! This draft specification is likely to change before being
+accepted as standard by the RISC-V Foundation.}
+
+In Sv39, Sv48, and Sv57, bits 62--61 of a leaf page table entry indicate the use
+of page-based memory types that override the PMA(s) for the associated memory
+pages.  The encoding for the PBMT bits is captured in Table~\ref{pbmt}.
+
+\begin{table*}[h!]
+\begin{center}
+\begin{tabular}{|r|l|}
+\hline
+Value  & Page-Based Memory Attributes \\
+\hline
+0      & None \\
+1      & Non-cacheable, idempotent, weakly-ordered (RVWMO or RVTSO), main memory \\
+2      & Non-cacheable, non-idempotent, strongly-ordered (channel 0), I/O \\
+3      & {\em Reserved for future standard use} \\
+\hline
+\end{tabular}
+\end{center}
+\caption{Encodings for the PBMT field in Sv39, Sv48, and Sv57 PTEs.  Attributes
+not mentioned are inherited from the PMA associated with the physical address.}
+\label{pbmt}
+\end{table*}
+
+\begin{commentary}
+Future extensions may provide more and/or finer-grained control over which PMAs
+can be overridden.
+\end{commentary}
+
+For non-leaf PTEs, bits 62--61 are reserved for future standard use and must be
+cleared by software for forward compatibility.
+
+If the underlying physical memory attribute for a page is I/O, then accesses to
+that page with PBMT=1 will obey RVWMO or RVTSO rather than I/O strong ordering
+rules, and accesses to such pages are considered main memory rather than I/O
+for the purposes of FENCE, {\em.aq}, and {\em.rl}.
+
+If the underlying physical memory attribute for a page is main memory, then
+accesses to that page via a page table entry with PBMT=2 obey strong channel 0
+ordering with respect to other accesses to physical main memory and to other
+accesses to pages with PBMT=2.  Furthermore, accesses to such pages are
+considered I/O rather than main memory for the purposes of FENCE, {\em.aq}, and
+{\em.rl}.
+
+With Svpbmt enabled, it is possible for multiple virtual aliases of the same
+physical page to exist simultaneously with different memory attributes.  It is
+also possible for a U-mode or S-mode mapping through a PTE with Svpbmt enabled
+to observe different memory attributes for a given region of physical memory
+than a concurrent access to the same page performed by M-mode or when {\tt
+satp}.MODE=Bare.  If accesses are performed simultaneously to the same region
+of memory using different attributes, there may be a loss of coherence and/or
```

Comment:
Since "accesses" here can be speculative, this should be changed to describe the architectural scenario, which is something about having conflicting valid mappings exist across all harts logically concurrently.  If SW robustly changes attributes with no overlap (e.g., some mutually globally invalid period), then is no platform-specific mechanism required?

I don't think that this is attempting to say that only explicit architectural accesses can cause this.